inst/templates/ad_adlb.R

```diff
@@ -0,0 +1,328 @@
+  "AST",     "AST",    "Aspartate Aminotransferase (U/L)", 5,
+  "BASO",    "BASO",   "Basophils (GI/L)", 6,
+  "BILI",    "BILI",   "Bilirubin (umol/L)", 7,
+  "BUN",     "BUN",    "Blood Urea Nitrogen (mmol/L)", 8,
+  "CA",      "CA",     "Calcium (mmol/L)", 9,
+  "CHOLES",  "CHOLES", "Cholesterol (mmol/L)", 10,
+  "CK",      "CK",     "Creatinine Kinase (U/L)", 11,
+  "CL",      "CL",     "Chloride (mmol/L)", 12,
+  "COLOR",   "COLOR",  "Color", 13,
+  "CREAT",   "CREAT",  "Creatinine (umol/L)", 14,
+  "EOS",     "EOS",    "Eosinophils (GI/L)", 15,
+  "GGT",     "GGT",    "Gamma Glutamyl Transferase (U/L)", 16,
+  "GLUC",    "GLUC",   "Glucose (mmol/L)", 17,
+  "HBA1C",   "HBA1C",  "Hemoglobin A1C (1)", 18,
+  "HCT",     "HCT",    "Hematocrit (1)", 19,
+  "HGB",     "HGB",    "Hemoglobin (mmol/L)", 20,
+  "K",       "POTAS",  "Potassium (mmol/L)", 21,
+  "KETONES", "KETON",  "Ketones", 22,
+  "LYM",     "LYMPH",  "Lymphocytes (GI/L)", 23,
+  "MACROCY", "MACROC", "Macrocytes", 24,
+  "MCH",     "MCH",    "Ery. Mean Corpuscular Hemoglobin (fmol(Fe))", 25,
+  "MCHC",    "MCHC",   "Ery. Mean Corpuscular HGB Concentration (mmol/L)", 26,
+  "MCV",     "MCV",    "Ery. Mean Corpuscular Volume (f/L)", 27,
+  "MICROCY", "MICROC", "Microcytes", 28,
+  "MONO",    "MONO",   "Monocytes (GI/L)", 29,
+  "PH",      "PH",     "pH", 30,
+  "PHOS",    "PHOS",   "Phosphate (mmol/L)", 31,
+  "PLAT",    "PLAT",   "Platelet (GI/L)", 32,
+  "POIKILO", "POIKIL", "Poikilocytes", 33,
+  "POLYCHR", "POLYCH", "Polychromasia", 34,
+  "PROT",    "PROT",   "Protein (g/L)", 35,
+  "RBC",     "RBC",    "Erythrocytes (TI/L)", 36,
+  "SODIUM",  "SODIUM", "Sodium (mmol/L)", 37,
+  "SPGRAV",  "SPGRAV", "Specific Gravity", 38,
+  "TSH",     "TSH",    "Thyrotropin (mU/L)", 39,
+  "URATE",   "URATE",  "Urate (umol/L)", 40,
+  "UROBIL",  "UROBIL", "Urobilinogen", 41,
+  "VITB12",  "VITB12", "Vitamin B12 (pmol/L)", 42,
+  "WBC",     "WBC",    "Leukocytes (GI/L)", 43
+)
+
+
+# ---- Derivations ----
+
+# Get list of ADSL vars required for derivations
+adsl_vars <- vars(TRTSDT, TRTEDT, TRT01A, TRT01P)
+
+adlb <- lb %>%
+
+  # Join ADSL with LB (need TRTSDT for ADY derivation)
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = vars(STUDYID, USUBJID)
+  ) %>%
+
+  # Calculate ADT, ADY
+  derive_vars_dt(
+    new_vars_prefix = "A",
+    dtc = LBDTC,
+    flag_imputation = FALSE
+  ) %>%
+
+  derive_var_ady(reference_date = TRTSDT, date = ADT)
+
+# Get WBC
+# Uncomment when derive_param_wbc_abs() is merged (PR #1035)
+# adlb <- adlb %>%
+#   derive_param_wbc_abs(#dataset = input,
+#                        by_vars = vars(USUBJID, VISIT),
+#                        set_values_to = vars(PARAMCD = "LYMPHSI",
+#                                           DTYPE = "CALCULATION"),
+#                        filter_diff = LBSTRESU == "fraction of 1",
+#                        wbc_code = "WBC",
+#                        diff_code = "LYMLE")
```

Comment:
I updated the setup for this function, so the code needs some updates. But let's wait for the function to be merged in devel in case there are further changes.

Reply:
Thank you @yangc71 ! Let's wait for the merging, sounds like a good plan.

Reply:
This part of the code has been removed as @konstantinakoukR found our dataset to be lacking